pkg/framework/plugins/nodeutilization/highnodeutilization.go

```diff
@@ -101,19 +101,51 @@ func (h *HighNodeUtilization) Balance(ctx context.Context, nodes []*v1.Node) *fr
 		}
 	}
 
-	sourceNodes, highNodes := classifyNodes(
-		getNodeUsage(nodes, h.usageClient),
-		getNodeThresholds(nodes, h.args.Thresholds, h.targetThresholds, h.resourceNames, false, h.usageClient),
-		func(node *v1.Node, usage NodeUsage, threshold NodeThresholds) bool {
-			return isNodeWithLowUtilization(usage, threshold.lowResourceThreshold)
+	nodesMap, nodesUsageMap, podListMap := getNodeUsageSnapshot(nodes, h.usageClient)
+	nodeThresholdsMap := getStaticNodeThresholds(nodes, h.args.Thresholds, h.targetThresholds)
+	nodesUsageAsNodeThresholdsMap := nodeUsageToResourceThresholds(nodesUsageMap, nodesMap)
+	nodeGroups := classifyNodeUsage(
+		nodesUsageAsNodeThresholdsMap,
+		nodeThresholdsMap,
+		[]classifierFnc{
+			// underutilized nodes
+			func(nodeName string, usage, threshold api.ResourceThresholds) bool {
+				return isNodeBelowThreshold(usage, threshold)
+			},
+			// every other node that is schedulable
+			func(nodeName string, usage, threshold api.ResourceThresholds) bool {
+				defer fmt.Printf("exiting classifyNodeUsage.classifier[1]\n")
+				if nodeutil.IsNodeUnschedulable(nodesMap[nodeName]) {
+					klog.V(2).InfoS("Node is unschedulable", "node", klog.KObj(nodesMap[nodeName]))
+					return false
+				}
+				return true
+			},
 		},
-		func(node *v1.Node, usage NodeUsage, threshold NodeThresholds) bool {
-			if nodeutil.IsNodeUnschedulable(node) {
-				klog.V(2).InfoS("Node is unschedulable", "node", klog.KObj(node))
-				return false
-			}
-			return !isNodeWithLowUtilization(usage, threshold.lowResourceThreshold)
-		})
+	)
+
+	// convert groups node []NodeInfo
+	nodeInfos := make([][]NodeInfo, 2)
+	category := []string{"underutilized", "overutilized"}
+	for i := 0; i < 2; i++ {
```

Comment:
My linter complains about this one, it says "for loop can be modernized using range over int". I think it means a `range` on `nodeGroups` (maybe `for i, nodeGroup := range nodeGroups` ?)